src/config/index.ts

```diff
@@ -0,0 +1,156 @@
+    vscode.window.showInformationMessage("CHANGED: ", e.fsPath);
+
+    if (vscode.workspace.workspaceFolders) {
+        for (let k = 0; k < vscode.workspace.workspaceFolders.length; k++) {
+            const f = vscode.workspace.workspaceFolders[k];
+            try {
+                let d = path.parse(e.fsPath).dir;
+                if (d === f.uri.fsPath) {
+                    workspaceConfig.set(f.uri.fsPath, parseConfig(f));
+                }
+            } catch {}
+            //vscode.window.showInformationMessage("FULL: " + f.uri.path + ": " + getContext(f));
+        }
+    }
+}
+
+let fsw = vscode.workspace.createFileSystemWatcher("**/.minetest-tools.json");
+fsw.onDidChange(updateConfigFromFileChanges);
+fsw.onDidCreate(updateConfigFromFileChanges);
+fsw.onDidDelete(updateConfigFromFileChanges);
+
+/**
+ * Parse the config file of the workspace folder if present
+ *
+ * Only for internal use inside this file
+ */
+function parseConfig(f: vscode.WorkspaceFolder): ExtensionConfigFolder | null {
+    const configPath = path.join(f.uri.fsPath, ".minetest-tools.json");
+    try {
+        const content = fs.readFileSync(configPath, { encoding: "utf-8" });
+        const parsedContent = JSON.parse(content);
+        if (parsedContent.context !== "game" && parsedContent.context !== "mod") {
+            return null;
+        } else {
+            return {
+                context: parsedContent.context,
+            };
+        }
+    } catch (error) {
+        return null;
+    }
+}
+
+/**
+ * Save a ExtensionConfigFolder to the config file in the root of the given workspace folder
+ *
+ * Only for internal use inside this file
+ */
+function saveConfig(f: vscode.WorkspaceFolder, c: ExtensionConfigFolder | null) {
+    const configPath = path.join(f.uri.fsPath, ".minetest-tools.json");
+    try {
+        //const content = fs.readFileSync(configPath, { encoding: "utf-8" });
+        //const parsedContent = JSON.parse(content);
+        fs.writeFileSync(configPath, JSON.stringify(c, undefined, 4));
+    } catch (error) {
+        return null;
+    }
+}
+
+/**
+ * Return the configuration for a workspace, or null if context
+ */
+function getConfig(w: vscode.WorkspaceFolder): ExtensionConfigFolder | null {
+    let c = workspaceConfig.get(w.uri.fsPath);
+    if (c === undefined) {
+        return null;
+    } else {
+        return c;
+    }
+}
+
+/**
+ * Get the configuration of the extension context or "none" for a {@linkcode vscode.WorkspaceFolder WorkspaceFolder}, with TypeScript annotations
+ */
+function getContext(w: vscode.WorkspaceFolder): "none" | ExtensionConfigContext {
+    const t = workspaceConfig.get(w.uri.fsPath)?.context;
+    return t ? t : "none";
+}
+
+/**
+ * (WIP) Update the configuration of the extension context for the {@linkcode vscode.WorkspaceFolder WorkspaceFolder}
+ *
+ * @todo should this function be able to actually delete the config file is called with "none"?
```

Comment:
@GreenXenith What is your opinion about this one? Maybe the JSON file could allow a "none" value?

Reply:
When the configuration file has more properties in it, then it would have to check to make sure the rest of the file is also empty before deleting it. Just allowing a "none" value feels simpler and avoids that extra logic.

Reply:
Ok, will change that.

Reply:
Fixed.